shared/actions/chat/inbox.js

```diff
@@ -2,141 +2,279 @@
 import * as ChatTypes from '../../constants/types/flow-types-chat'
 import * as Constants from '../../constants/chat'
 import * as Creators from './creators'
-import * as Saga from '../../util/saga'
 import {List, Map} from 'immutable'
 import {TlfKeysTLFIdentifyBehavior} from '../../constants/types/flow-types'
-import {call, put, select, race} from 'redux-saga/effects'
+import {call, put, select, race, fork} from 'redux-saga/effects'
 import {delay} from 'redux-saga'
 import {parseFolderNameToUsers} from '../../util/kbfs'
 import {requestIdleCallback} from '../../util/idle-callback'
+import {singleFixedChannelConfig, takeFromChannelMap} from '../../util/saga'
 import {unsafeUnwrap} from '../../constants/types/more'
 import {usernameSelector} from '../../constants/selectors'
+import {isMobile} from '../../constants/platform'
 
 import type {SagaGenerator, ChannelMap} from '../../constants/types/saga'
-import type {TypedState} from '../../constants/reducer'
 
-const _metaDataSelector = (state: TypedState) => state.chat.get('metaData')
-const followingSelector = (state: TypedState) => state.config.following
+type UntrustedState = 'unloaded' | 'loaded' | 'loading'
 
-let _loadedInboxOnce = false
-function * onLoadInboxMaybeOnce (action: Constants.LoadInbox): SagaGenerator<any, any> {
-  if (!_loadedInboxOnce || action.payload.force) {
-    _loadedInboxOnce = true
-    yield call(onLoadInbox)
+// Common props for getting the inbox
+const _getInboxQuery = {
+  computeActiveList: true,
+  readOnly: false,
+  status: Object.keys(ChatTypes.CommonConversationStatus).filter(k => !['ignored', 'blocked'].includes(k)).map(k => ChatTypes.CommonConversationStatus[k]),
+  tlfVisibility: ChatTypes.CommonTLFVisibility.private,
+  topicType: ChatTypes.CommonTopicType.chat,
+  unreadOnly: false,
+}
+
+let _inboxUntrustedState: UntrustedState = 'unloaded'
+let _inboxUntrustedError = null
+
+// Load the inbox if we haven't yet, mostly done by the UI
+function * onInitialInboxLoad (action: Constants.LoadInbox): SagaGenerator<any, any> {
+  if (_inboxUntrustedState === 'loading') {
+    return
+  }
+
+  const {force} = action.payload
+  if (_inboxUntrustedState === 'unloaded' || _inboxUntrustedError || force) {
+    _inboxUntrustedState = 'loading'
+    _inboxUntrustedError = null
+    yield call(onInboxStale)
+    if (!isMobile) {
+      yield fork(_backgroundUnboxLoop)
+    }
   }
 }
 
-function * onLoadInbox (): SagaGenerator<any, any> {
-  const channelConfig = Saga.singleFixedChannelConfig([
-    'chat.1.chatUi.chatInboxUnverified',
-    'chat.1.chatUi.chatInboxConversation',
-    'chat.1.chatUi.chatInboxFailed',
-    'finished',
-  ])
+// On desktop we passively unbox inbox items
+function * _backgroundUnboxLoop () {
+  const maxPerLoop = 10
+
+  while (true) {
+    yield call(delay, 10 * 1000)
+    const inboxes = yield select(state => state.chat.get('inbox'))
+
+    const conversationIDKeys = []
+    inboxes.forEach(inbox => {
```

Comment:
also has to map so its like `const conversationIDKeys = inboxes.filter(i => i.state === 'untrusted').take(maxPerLoop).map(i => i.conversationIDKey).toArray()
`